src/daemon/dlt-daemon.c

```diff
@@ -1000,7 +1007,7 @@ int dlt_daemon_local_init_p2(DltDaemon *daemon, DltDaemonLocal *daemon_local, in
                         daemon_local->flags.contextLogLevel,
                         daemon_local->flags.contextTraceStatus,
                         (daemon_local->flags.offlineTraceDirectory[0] !=
-                         '\0') ? DLT_USER_MODE_BOTH : DLT_USER_MODE_EXTERNAL,
+                         '\0') && daemon_local->flags.offlineTraceOn!=0 ? DLT_USER_MODE_BOTH : DLT_USER_MODE_EXTERNAL,
```

Comment:
Please add the white space before and after "!="

Reply:
Done